source/common/http/http1/codec_impl.h

```diff
@@ -317,6 +317,8 @@ class ConnectionImpl : public virtual Connection, protected Logger::Loggable<Log
 
   static http_parser_settings settings_;
 
+  bool dispatching_ : 1;
+  absl::Status codec_exception_;
```

Comment:
Can you comment on the use of this new field?

We should also ASSERT(codec_exception_.ok()) in various places including at the top of dispatchSlice and places where we end up overwriting it.

Reply:
I think with custom type we can push this logic into the type and either make it ASSERT when an error condition is being overwritten or make it idempotent.

Reply:
Some methods shouldn't be entered when there's an error already set.  I agree that we can have the custom type complain when an error value is overwritten.